src/mongo_client.ts

```diff
@@ -364,6 +369,32 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         return true;
       }
     };
+    this.checkForNonGenuineHosts();
+  }
+
+  /** @internal */
+  private checkForNonGenuineHosts() {
+    if (this.mongoLogger && this.mongoLogger.logDestination) {
+      const documentDBHostnames = this[kOptions].hosts.filter((hostAddress: HostAddress) =>
+        isHostMatch(DOCUMENT_DB_CHECK, hostAddress.host)
+      );
+
+      const srvHostIsDocumentDB = isHostMatch(DOCUMENT_DB_CHECK, this[kOptions].srvHost);
+
+      if (documentDBHostnames.length !== 0 || srvHostIsDocumentDB) {
+        this.mongoLogger.info('client', DOCUMENT_DB_MSG);
+      }
+
+      const cosmosDBHostnames = this[kOptions].hosts.filter((hostAddress: HostAddress) =>
+        isHostMatch(COSMOS_DB_CHECK, hostAddress.host)
+      );
+
+      const srvHostIsCosmosDB = isHostMatch(COSMOS_DB_CHECK, this[kOptions].srvHost);
+
+      if (cosmosDBHostnames.length !== 0 || srvHostIsCosmosDB) {
+        this.mongoLogger.info('client', COSMOS_DB_MSG);
+      }
+    }
```

Comment:
Would it ever be the case that we would match both CosmosDB and DocumentDB? If not, I think we should make that clear with the control flow by making the two calls to `this.mongoLogger.info` live in separate branches of an if statement. Otherwise, disregard.

Reply:
If you find one and print the message that should be it. Though it shouldn't ever be possible to match both based on the regex, even if it were the case only one message would need to be displayed.

Reply:
it is "technically" possible to match both CosmosDB and DocumentDB, but extremely unlikely, I'll refactor these to live in separate branches. We also want to comply with [DRIVERS-2583](https://jira.mongodb.org/browse/DRIVERS-2583)'s "Drivers should not emit more than one log message per MongoClient construction" directive, and not allow more than one log message per client anyway.